### ISSUE TITLE
Delay on('notification')

### ISSUE THREAD

**Issue by dmarcus1** [phonegap/phonegap-plugin-push#1700]
This plugin works well. I'm able to receive push notifications. I just have a question about the usage.

Is there a way to delay firing on('notification') until another callback is complete? My other callback checks facebook to see if I'm logged in. Once that's finished, I'd like on('notification') to fire.

### Expected Behaviour
I want on('notification') to fire only after I verify that I'm logged into facebook api.

### Actual Behaviour
on('notification') sometimes gets called before the function that verifies that I'm logged into facebook. Sometimes the function that verifies that I'm logged in gets called before on('notification').

### Reproduce Scenario (including but not limited to)
This isn't a bug. I'm just asking about the functionality of this plugin.

#### Steps to Reproduce 
Create a facebook api login key. Call their function to see if you're logged in. Only call on('notification') once it verifies that you're logged in.

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
iOS 10.1.3

#### (Android) What device vendor (e.g. Samsung, HTC, Sony...)
iPhone

#### Cordova CLI version and cordova platform version

    6.1.0

#### Plugin version

    1.9.0

#### Sample Push Data Payload

{ "data": { "title": "Hello", "message": "World"} }

#### Sample Code that illustrates the problem

push.on('notification', function(data) {
	dosomething1();
});

document.addEventListener("deviceready", function(){
    checkfacebooklogin(status) {
    if(status == 1) {
        dosomething2();
     }
  }
});

a status of 1 means you're logged in. If you open the app by clicking on a push notification then dosomething1() should get called and dosomething2() should not get called. If you open the app and there's no push notification then dosomething2() should get called and dosomething1() should not get called. I don't think I can just put on('notification') inside of if(status == 1).

#### Logs taken while reproducing problem
I don't have any logs

**Comment by fredgalvao** [comment]
A simple way would be to only call `.init()` after you log the user in.

Another one would be to do proper handling through either one of these:
- callback dependency hell
- local caching of a notification event
- joining promises

All of those options mentioned right above relate to javascript skills, and quite nothing to do with this plugin per se.

**Comment by davemarc** [comment]
@fredgalvao thanks for getting back to me. I'll look into the javascript solutions you suggested. I just wanted to see if there was a built-in way in the plugin. 

I didn't explain my question well the first time. So let me try again. My problem happens when my phone gets a push notification, and I click on the push notification to open the app. It's too bad there isn't a way to say 

if(!receivednotification) {
  function1()
}
else {
  function2();
}

The problem is that I don't know when on('notification') gets called. In the case where my phone gets a push notification, and I click the push notification to open the app, on('notification') does fire shortly after opening the app. But there's no way to tell exactly when it fires. So I can't wrap code in an if statement, if(!receivednotification) {function1()}, that I don't want to run if the app is opened by clicking on a push notification.

I want function1() to run only if I didn't just open the app by clicking on a push notification. function2() should run only if I open the app by clicking on a push notification.

Right now I have something in like:

function1();
push.on('notification') {
   function2();
}

function1() gets called always, and then function2() gets called in addition to function1() if I opened up the app by clicking the notification. I only want function2() to get called if I get a notification. I wish I could just break out of function1() as soon as I get a notification. Or better yet not even call function1() if I just opened the app by clicking on a push notification.

**Comment by fredgalvao** [comment]
Now that you've explained your point in detail, I can tell you that this is a feature request that I myself have made/mentioned multiple times.

Please read [this comment](https://github.com/phonegap/phonegap-plugin-push/issues/501#issuecomment-171519136) and [this other comment](https://github.com/phonegap/phonegap-plugin-push/issues/501#issuecomment-250630474) with the surrounding context.
It was mentioned again [here](https://github.com/phonegap/phonegap-plugin-push/issues/1265#issuecomment-254259011).

If you think that the #501 issue is the same thing you want, then I'd suggest you
- close this issue as a duplicate
- 👍 that issue
- comment there to reinforce it, making sure you try and put your needs on it as clearly as you can

**Comment by dmarcus1** [comment]
Okay that sounds good. I'll comment there over the weekend. Thanks again for your help.
